components/download/internal/common/download_item_impl.cc

```diff
@@ -1774,7 +1774,13 @@ void DownloadItemImpl::OnDownloadTargetDetermined(
     return;
   }
 
-  destination_info_.target_path = target_path;
+  // Neeva(kobec): Rename target_path in case there is already a file with the same name. 
+  base::FilePath new_target_path = base::GetUniquePath(target_path);
```

Comment:
I'm still curious if this is the best approach. Since desktop Chrome solves the issue somehow without needing this code change, it makes me curious how and if that would point us to a better solution.

Another issue perhaps negligible issue I see is that this is doing synchronous file IO. Normally, in Chromium code we try to avoid doing that on the main thread, which is where I believe this code is running. (Correct me if I'm wrong about that, but most code in Chromium runs on the main thread.)

In a debug build, this would probably trip up an assertion about doing synchronous blocking IO on the main thread. We may want to use `base::ScopedAllowBlocking` if needed to workaround this issue.

But again, maybe the desktop way already handles this whole issue well in a non-blocking way.

Reply:
On the synchronous file IO: 
https://source.chromium.org/chromium/chromium/src/+/main:components/download/internal/common/download_item_impl.cc?q=OnDownloadTargetDetermined

I think that the function running the `GetUniquePath` code isn't on the main thread since its posted.

Reply:
`RenameAndUniquify()` is called[ later in the function on another thread](https://source.chromium.org/chromium/chromium/src/+/main:components/download/internal/common/download_item_impl.cc;l=1793?q=OnDownloadTargetDetermined) and `base::GetUniquePath(new_path)` is eventually called on that [same thread](https://source.chromium.org/chromium/chromium/src/+/main:components/download/internal/common/download_file_impl.cc;l=411;drc=f97e7e130b02a0fee5a06aa9cdf25d3a0a3715d0?q=download_file_impl.cc&ss=chromium%2Fchromium%2Fsrc).

Reply:
So I'm kind of sure that synchronous file IO operations on a background thread is a negligible issue (because it's already happening in their code)

Reply:
On seeing if there's a desktop way, I'll dig deeper on that

Reply:
> and base::GetUniquePath(new_path) is eventually called on that same thread

Agreed. Nice investigation! Thank you.